server/routes/blog.service.js

```diff
@@ -25,29 +26,50 @@ const getBlogByCategory = async (category) => {
 
 // mutations
 const createBlog = async (blogData) => {
+
+    const validatedBlog = BlogSchema.parse(blogData)
+
     return db.blog.create({
-        data: {
-            title: blogData.title,
-            category: blogData.category,
-            description: blogData.description,
-            content: blogData.content,
-            imageUrl: blogData.imageUrl,
-            authorId: blogData.authorId,
-            datePublished: blogData.datePublished,
-        },
+        data:validatedBlog,
         select: {
             id: true,
         },
     })
 }
 
 const updateBlog = async (id, blogData) => {
+    const validatedData = BlogSchema.parse(blogData)
     return db.blog.update({
         where: { id: String(id) },
-        data: blogData,
+        data: validatedData,
     })
 }
 
+// const createBlog = async (blogData) => {
+//     return db.blog.create({
+//         data: {
+//             title: blogData.title,
+//             category: blogData.category,
+//             description: blogData.description,
+//             content: blogData.content,
+//             imageUrl: blogData.imageUrl,
+//             authorId: blogData.authorId,
+//             datePublished: blogData.datePublished,
+//         },
+//         select: {
+//             id: true,
+//         },
+//     })
+// }
+
+
+// const updateBlog = async (id, blogData) => {
+//     return db.blog.update({
+//         where: { id: String(id) },
+//         data: blogData,
+//     })
+// }
+
 const deleteBlog = async () => {
     return db.blog.delete({
         where: { id: String(id) },
```

Comment:
We also have to make sure not to delete the blog without making sure that the request is being sent by the person who created the blog. Otherwise, anyone can send a backend-only request (through postman) with the right data and the post will get deleted. So, the deletion would look something like this
```js
... await db.blog.delete({
     where: {
       id: validatedBlog.data.id,
       authorId: currentUser.id,
     }
   })
```

Reply:
I would rather create a boolean flag around delete blog so we can at least retain the data, what do you think?

Reply:
That's a good idea! That's what I do for my other websites whenever comments are deleted, it's a good idea to keep some things up in case something needs to be brought up in the future.